src/uri/uri.cc

```diff
@@ -249,6 +249,54 @@ auto URI::path() const -> std::optional<std::string> {
   return path_;
 }
 
+auto URI::path(const std::string &path) -> URI & {
+  if (path.empty()) {
+    this->path_ = std::nullopt;
+    return *this;
+  }
+
+  auto is_relative = path.starts_with("./") || path.starts_with("../");
+  if (is_relative) {
+    throw URIError{
+        "You cannot set a relative path, use URI::resolve_from() instead"};
```

Comment:
```suggestion
        "You cannot set a relative path"};
```

I guess we can just explain the issue without specific instructions, otherwise the error message won't make much sense outside of a C++ context